apis/Google.Cloud.Firestore/Google.Cloud.Firestore/DocumentSet.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Copyright 2018, Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+
+namespace Google.Cloud.Firestore
+{
+    /// <summary>
+    /// Immutable set of DocumentSnapshots, ordered by a custom comparer.
+    /// </summary>
+    internal sealed class DocumentSet : IEnumerable<DocumentSnapshot>
+    {
+        // TODO: Investigate whether the immutable collections are okay here. ImmutableSortedSet uses an AVL tree,
+        // not a red-black tree.
```

Comment:
Both are balanced binary trees, so it's fine.

Reply:
(Will remove the comment at a later date :)